bot.ml

```diff
@@ -672,6 +690,34 @@ let callback _conn req body =
       | Ok (_, GitHub_subscriptions.RemovedFromProject _) ->
           Server.respond_string ~status:`OK
             ~body:"Note card removed from project: nothing to do." ()
+      | Ok (_, GitHub_subscriptions.CommentCreated comment_info)
+        when string_match ~regexp:"@coqbot: minimize[^```]*```\\([^```]+\\)```"
+               comment_info.body ->
+          let comment_thread_id =
+            match comment_info.pull_request with
+            | None ->
+                comment_info.issue.id
+            | Some pr ->
+                pr.issue.id
+          in
+          let script = Str.matched_group 1 comment_info.body in
+          (fun () ->
+            Lwt_io.printf "Found script: %s\n" script
+            >>= fun _ ->
+            git_coq_bug_minimizer ~script ~comment_thread_id
+              ~comment_author:comment_info.author ~bot_name ~bot_domain
+              ~token:github_access_token
+            >>= fun _ ->
+            GitHub_mutations.post_comment ~id:comment_thread_id
+              ~message:
+                (f
+                   "Hey @%s, the coq bug minimizer is running your script, \
+                    I'll come back to you with the results once it's done."
+                   comment_info.author)
+              ~token:github_access_token
```

Comment:
It would be best to not ignore the result of the previous action, and in particular, to not post this comment if the coq bug minimizer is not actually running anything because something went wrong. More generally, it would be best to use this `fun _ ->` pattern as little as possible. For instance, the first of the three here can simply be replaced by `fun () ->` I think.